qiskit/primitives/base/base_primitive.py

```diff
@@ -73,14 +73,16 @@ def _validate_circuits(
 
     @staticmethod
     def _validate_parameter_values(
-        parameter_values: Sequence[Sequence[float]] | Sequence[float] | float | None,
-        default: Sequence[Sequence[float]] | Sequence[float] | None = None,
+        parameter_values: Sequence[Sequence[float] | Mapping[ParameterExpression, float]]
+        | Sequence[float]
+        | Mapping[ParameterExpression, float]
+        | float
+        | None,
+        circuits: tuple[QuantumCircuit, ...],
```

Comment:
I would pass a `Sequence[Sequence[ParameterExpresion]]` here instead. No need for the entire circuit. 

Also, inputs should remain as general as possible, so `Sequence` is preferred over `tuple` for _inputs_; whereas `tuple` is preferred for _outputs_.

Reply:
> I would pass a Sequence[Sequence[ParameterExpresion]] here instead.

Great suggestion. Thanks.